File: backend/src/main/java/com/zosh/dto/ChatDto.java
package com.zosh.dto;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ChatDto {

	
	private Long id;
    private MessageType type; // 메시지 타입
    private String roomId; // 방 번호
    private String sender; // 채팅을 보낸 사람
    private String email; //보낸 사람 이메일(프로필 사진 추출용)
    private String message; // 메시지
    private LocalDateTime time; // 채팅 발송 시간간
    public enum MessageType{
        ENTER, TALK
    }
}

File: Kotlin/app/src/main/java/com/hippoddung/ribbit/network/bodys/chatbodys/ChatDto.kt
package com.hippoddung.ribbit.network.bodys.chatbodys

import android.os.Build
import androidx.annotation.RequiresApi
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.JsonPrimitive
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

@RequiresApi(Build.VERSION_CODES.O)
data class ChatDto constructor(
    val id: Long? = null,
    val type: MessageType = MessageType.TALK,   // 서버에서 구현하려다가 그만둔 기능
    val roomId: String? = null,
    val sender: String? = null,
    val email: String? = null,
    val message: String = "",
    val time: String? = null
) {
    enum class MessageType {
        ENTER, TALK
    }

    fun toJsonObject(): JsonObject {
        return JsonObject(
            mapOf(
                "id" to JsonPrimitive(id),
                "type" to JsonPrimitive(type.name),
                "roomId" to JsonPrimitive(roomId),
                "sender" to JsonPrimitive(sender),
                "email" to JsonPrimitive(email),
                "message" to JsonPrimitive(message),
                "time" to JsonPrimitive(time)
            )
        )
    }
}

